src/ds/d500/d500-color.cpp

```diff
@@ -0,0 +1,176 @@
+
+        std::unique_ptr<frame_timestamp_reader> ds_timestamp_reader_backup(new ds_timestamp_reader(backend.create_time_service()));
+        std::unique_ptr<frame_timestamp_reader> ds_timestamp_reader_metadata(new ds_timestamp_reader_from_metadata(std::move(ds_timestamp_reader_backup)));
+
+        auto enable_global_time_option = std::shared_ptr<global_time_option>(new global_time_option());
+        auto raw_color_ep = std::make_shared<uvc_sensor>("Raw RGB Camera",
+            backend.create_uvc_device(color_devs_info.front()),
+            std::unique_ptr<frame_timestamp_reader>(new global_timestamp_reader(std::move(ds_timestamp_reader_metadata), _tf_keeper, enable_global_time_option)),
+            this);
+
+        auto color_ep = std::make_shared<d500_color_sensor>(this,
+            raw_color_ep,
+            d500_color_fourcc_to_rs2_format,
+            d500_color_fourcc_to_rs2_stream);
+
+        color_ep->register_option(RS2_OPTION_GLOBAL_TIME_ENABLED, enable_global_time_option);
+
+        color_ep->register_info(RS2_CAMERA_INFO_PHYSICAL_PORT, color_devs_info.front().device_path);
+
+        _color_device_idx = add_sensor(color_ep);
+    }
+
+    void d500_color::init()
+    {
+        auto& color_ep = get_color_sensor();
+        auto& raw_color_ep = get_raw_color_sensor();
+
+        _ds_color_common = std::make_shared<ds_color_common>(raw_color_ep, color_ep,
+             _fw_version, _hw_monitor, this);
+
+        register_options();
+        register_metadata();
+        register_processing_blocks();
+    }
+
+    void d500_color::register_options()
+    {
+        auto& color_ep = get_color_sensor();
+        auto& raw_color_ep = get_raw_color_sensor();
+
+        _ds_color_common->register_color_options();
+
+        color_ep.register_pu(RS2_OPTION_AUTO_EXPOSURE_PRIORITY);
+
+        _ds_color_common->register_standard_options();
+
+        color_ep.register_pu(RS2_OPTION_HUE);
+    }
+
+    void d500_color::register_metadata()
+    {
+        auto& color_ep = get_color_sensor();
+
+        auto md_prop_offset = offsetof(metadata_raw, mode) +
+            offsetof(md_rgb_mode, rgb_mode) +
+            offsetof(md_rgb_normal_mode, intel_rgb_control);
+
+        color_ep.register_metadata(RS2_FRAME_METADATA_AUTO_EXPOSURE, make_attribute_parser(&md_rgb_control::ae_mode, md_rgb_control_attributes::ae_mode_attribute, md_prop_offset,
+            [](rs2_metadata_type param) { return (param != 1); })); // OFF value via UVC is 1 (ON is 8)
+
+        _ds_color_common->register_metadata();
+    }
+
+    void d500_color::register_processing_blocks()
+    {
+        auto& color_ep = get_color_sensor();
+
+        color_ep.register_processing_block(processing_block_factory::create_id_pbf(RS2_FORMAT_RAW16, RS2_STREAM_COLOR));
+
+        color_ep.register_processing_block(processing_block_factory::create_pbf_vector<m420_converter>(RS2_FORMAT_M420, map_supported_color_formats(RS2_FORMAT_M420), RS2_STREAM_COLOR));
+    }
```

Comment:
@maloel  need to make sure the input format on public d500

Reply:
Is there some open issue here?

Reply:
No open issue -- noted.